doc/user_guide/jupyter_chart.rst

```diff
@@ -0,0 +1,403 @@
+        color=alt.condition(brush, 'Origin:N', alt.value('grey')),
+    ).add_params(brush)
+
+    widget = alt.ChartWidget(chart)
+    widget
+
+.. raw:: html
+
+    <video controls>
+      <source src="https://github.com/altair-viz/altair/assets/15064365/2960fc28-885e-40a4-8bd5-9e4b56616afe">
+      Your browser does not support the video tag.
+    </video>
+
+
+.. warning::
+    The indices returned will only correspond to the input DataFrame for charts that do not include
+    aggregations. If a chart includes aggregations, then the ``alt.selection_point`` specification
+    should include either a ``fields`` or ``encodings`` argument, which will result in the
+    ``JupyterChart`` containing a ``PointSelection`` rather than an ``IndexSelection``.
+
+
+Interval Selections
+~~~~~~~~~~~~~~~~~~~
+The ``IntervalSelection`` class is used to store the current state of a Vega-Altair interval selection
+(as created by ``alt.selection_interval()``). In this case, the ``value`` property of the selection
+is a dictionary from column names to selection intervals
+
+.. code-block:: python
+
+    import altair as alt
+    from vega_datasets import data
+
+    source = data.cars()
+    brush = alt.selection_interval(name="interval")
+
+    chart = alt.Chart(source).mark_point().encode(
+        x='Horsepower:Q',
+        y='Miles_per_Gallon:Q',
+        color=alt.condition(brush, 'Cylinders:O', alt.value('grey')),
+    ).add_params(brush)
+
+    jchart = alt.JupyterChart(chart)
+    jchart
+
+
+.. raw:: html
+
+    <video controls>
+      <source src="https://github.com/altair-viz/altair/assets/15064365/8b1b6ea6-6d9d-4a53-aac9-3e120609d99d">
+      Your browser does not support the video tag.
+    </video>
+
+
+Observing Selections
+~~~~~~~~~~~~~~~~~~~~
+As with variable parameters, it's possible to register a callback function to be invoked
+when a selection changes by using the ``observe`` method on the ``selections`` property.
+Here is an example that listens for changes to an interval selection, then uses the selection
+value to filter the input DataFrame and display it's HTML representation. An ipywidgets ``VBox``
+is used to combine the chart and HTML table in a column layout.
+
+.. code-block:: python
+
+    import ipywidgets
+    from IPython.display import display
+    from ipywidgets import HTML, VBox
+
+    import altair as alt
+    from vega_datasets import data
+
+    source = data.cars()
+    brush = alt.selection_interval(name="brush")
+
+    chart_widget = alt.JupyterChart(alt.Chart(source).mark_point().encode(
+        x='Horsepower:Q',
+        y='Miles_per_Gallon:Q',
+        color=alt.condition(brush, 'Cylinders:O', alt.value('grey')),
+    ).add_params(brush))
+
+    table_widget = HTML(value=source.iloc[:0].to_html())
+
+    def on_select(change):
+        sel = change.new.value
+        if sel is None or 'Horsepower' not in sel:
+            filtered = source.iloc[:0]
+        else:
+            filtered = source.query(
```

Comment:
Can we build up this query as a separate step? It's a very interesting part of the new functionality and it deserves to be highlighted.

Reply:
Done in [56c9fe0](https://github.com/altair-viz/altair/pull/3137/commits/56c9fe09629730857b76d734b7f5621fb7c52073)

Reply:
I'm sorry for not being more clear. I meant to introduce this query also, or firstly, in the section prior 'Observe Selections'. It is interesting how one can (manually) extract data from the source dataframe using the interval selection.

Just wondering, would this still work with an interval selection on a chart including an  aggregation in combination with `alt.Chart().transformed_data()`?

Reply:
Ahh, yeah, that makes sense. I'll look at adding sections for filtering input DataFrame based on the point and interval selections (The index selection example already shows how to do this with `iloc`).

> Just wondering, would this still work with an interval selection on a chart including an aggregation in combination with alt.Chart().transformed_data()?

Are you wanting to filter that data after aggregation? If so, this should work with charts that include aggregation but no  binning (e.g. an ordinal or nominal bar chart with aggregation). For charts that include `timeunit` or `bin` binning, this might not work because the column resulting from `transformed_data` will generally have a different name. e.g. `column_start` instead of `column`. Though in this case the selection might use the renamed columns as well (I'm not sure off hand).

In any case, I think the generally more useful thing is to apply the selection filter to the pre-aggregated DataFrame and then return it (rather than applying the aggregations). This way the result corresponds, row for row, to the input DataFrame.

I need to think through it more carefully, but I'm picturing adding a `chart.selected_data` method that would input a list of selections. It would need to intelligently crawl through the chart's transform pipeline and apply the early transforms that produce any columns needed by the selection (e.g. the binning outputs), then apply the selection as a filter, then return the result without applying the downstream aggregations.  Or, maybe there is an option of whether or not to apply the downstream aggregations.

This would essentially automate the process of building the right pandas query string, handling cases where binned columns are renamed, and it would perform the filter in VegaFusion so it would be done by VegaFusion's current backend (e.g. DataFusion or DuckDB).  I'll open an issue in the VegaFusion repo on this at some point. But in the near term, we should document how to do the filtering with pandas.

Reply:
I added info on creating pandas query strings in [2ee9d76](https://github.com/altair-viz/altair/pull/3137/commits/2ee9d765d356a27711f6f56851c53c0a6fc3fcea). Does that look sufficient for now @mattijn?